### ISSUE TITLE
JsonApiResponseTrait::addHeaders behaviour

### ISSUE THREAD

**Issue by lunika** [nilportugues/symfony-jsonapi#30]
Hi, 

I would like to understand what it is the behaviour for the method `JsonApiResponseTrait::addHeaders` ?

This method do nothing more than returning the response parameter and is named `addHeaders` but there is no header parameter, only the response. Is this method only here to be overrided ? Something else ?

**Comment by strayobject** [comment]
Yeah, you should overwrite it in your class which is implementing JsonApiResponseTrait.
Remember that the response is immutable and as such your implementation should look along these lines:
```
protected function addHeaders(ResponseInterface $response)
    {
        $response = $response->withHeader('x-my-header', 'some value');
        
        return $response;
    }
```

An alternative to that would be to add the headers you need in the controller method (this way you are using symfony's param bag, for example:

```
public function myAction(Request $request)
    {
        [...]
        $response = $this->response($data);
        $response->headers->add(['x-my-header' => 'some value']);

        return $response;
    }
```
